core/state_transition.go

```diff
@@ -302,10 +320,11 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 		sender           = vm.AccountRef(msg.From())
 		rules            = st.evm.ChainConfig().Rules(st.evm.Context.BlockNumber, st.evm.Context.Random != nil)
 		contractCreation = msg.To() == nil
+		eip3860          = st.evm.Config.HasEip3860()
```

Comment:
We should either make `Is3860` part of the `Rules`, or make it part of `IsShanghai`. However, that means we'll lose the ability to do e.g. `Merge+3860`, but maybe that's ok

Reply:
I'd prefer to enable it in Shanghai as well although this may conflict with EOF testing which must be done on top of EIP-3860. Is `Shanghai+3540` an option for EOF?